### DISCUSSION TITLE
[apex] What is the syntax to declare multiple params in comment section of a method

### DISCUSSION THREAD

**Discussion by ssc-crm-git-user** [pmd/pmd#3109] [Q&A]
What is the syntax to declare multiple params in comment section of a method. I have used below format

```
   @description  Overloaded Method to log a single record to the application log
   @param        logLevel 
   @param        sourceClass 
   @param        sourceFunction
   @param        referneceId
   @param        referenceInfo
   @param        logMessage
   @param        payLoad
   @param        ex
   @param        timeTaken
```

its exactly following below method sequence, but still I am getting '**Missing or mismatched ApexDoc @param**' violation

 public static void logMessage(String logLevel, String sourceClass, String sourceFunction, String referenceId, String referenceInfo, String logMessage, String payLoad, Exception ex, long timeTaken) {

**Comment by adangel** [comment]
Your syntax looks correct. However, there is a typo in your apexdoc: referneceId vs. referenceId.

I guess, we should improve the message of the rule to explicitly say, which parameter is missing/mismatched. While looking at the code right now, I noticed, that we compare the param names case sensitive. As far as I know, this should be case-insensitive.

Please create a new issue with these improvements.
